### PR TITLE
Magento 2 Cookbook

### PR DESCRIPTION
**Pull request by jerrywardlow**
This pull request creates a cookbook which installs Magento 2. This has been a long work in progress and all requests, revisions, opinions, personal vendettas, criticisms, and general taunts are greatly appreciated.

Enhancements for the next release include the break-out of the `robots`, `cli-tools`, `data-bag`, and `users` recipes into individual cookbooks.

This cookbook includes a set of passing Kitchen tests

```
-----> Verifying <default-centos-7>
...
-----> Running serverspec test suite
...
       31 examples, 0 failures

       Finished verifying <default-centos-7> (0m6.10s).
```

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#
+# Cookbook Name:: magento
+# Recipe:: default
+#
+
+include_recipe 'cop_magento::data-bag'
+include_recipe 'cop_magento::users'
+include_recipe 'cop_magento::cli-tools'
+include_recipe 'cop_magento::dir-structure'
+if node.chef_environment == "development"
```
Comment (hardyjohnson) [inline recipes/default.rb]: What was the argument for this vs environment attributes feeding into dir-structure?

File: test/integration/environments/development.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                },
+                "fpm": {
+                    "ini": {
+                        "display_errors": "1",
+                        "error_log": "/var/log/php-fpm.error.log",
+                        "mail_add_x_header": "1",
+                        "report_zend_debug": "1",
+                        "sendmail_path": "/usr/sbin/sendmail -t -i",
+						"disable_functions": "dl, exec, gc_collect_cycles, gc_disable, gc_enable, gc_enabled, get_cfg_var, get_current_user, getlastmod, getmygid, getmyinode, getmyuid, lchgrp, lchown, link, passthru, pclose, pcntl_alarm, pcntl_exec, pcntl_fork, pcntl_get_last_error, pcntl_getpriority, pcntl_setpriority, pcntl_signal, pcntl_signal_dispatch, pcntl_sigprocmask, pcntl_sigtimedwait, pcntl_sigwaitinfo, pcntl_strerror, pcntl_wait, pcntl_waitpid, pcntl_wexitstatus, pcntl_wifexited, pcntl_wifsignaled, pcntl_wifstopped, pcntl_wstopsig, pcntl_wtermsig, php_uname, popen, posix_getlogin, posix_mkfifo, posix_ttyname, proc_close, proc_get_status,proc_nice, proc_open, proc_terminate, prog_get_status, set_time_limit, shell_exec, show_source, system"
+                    },
```
Comment (mfdj) [inline test/integration/environments/development.json]: `disable_functions` whitespace funky

File: templates/composer/composer.json.erb [hunk 3]
```diff
@@ -0,0 +1,67 @@
+{
+    "name": "magento/copious-dev-box",
+    "description": "eCommerce Platform for Growth (<%= node['magento']['edition'] %> Edition)",
+    "type": "project",
+    "license": [
+        "proprietary"
+    ],
+    "repositories": [
```
Comment (mfdj) [inline templates/composer/composer.json.erb]: I'm reminded of a suggestion when working with the xerox project: make `name` and `description` configurable attributes so all of our projects don't report identical information (for instance `composer info --self` will report these, happens in some other composer spots as well)

File: test/integration/default/serverspec/default_spec.rb [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        it { should be_mode 644 }
+    end
+
+    describe file('/var/www/example.com/shared/composer/composer.json') do
+        it { should exist }
+        it { should be_owned_by 'www-data' }
+        it { should be_grouped_into 'www-data' }
+        it { should be_mode 644 }
+    end
+end
```
Comment (mfdj) [inline test/integration/default/serverspec/default_spec.rb]: 😍 😍 😍 😍 😍 😍 😍 😍 😍 😍

File: templates/composer/composer.json.erb (hunk 3 above)
Comment (jerrywardlow) [inline templates/composer/composer.json.erb]: 🥇 @mfdj

**Pull request merged**